pkg/sql/show_fingerprints.go

```diff
@@ -176,41 +213,54 @@ func (n *showFingerprintsNode) nextTenant(params runParams) (bool, error) {
 	if err != nil {
 		return false, err
 	}
+	n.run.values[0] = tree.NewDString(string(tinfo.Name))
 
 	tid, err := roachpb.MakeTenantID(tinfo.ID)
 	if err != nil {
 		return false, err
 	}
 
+	// We want to write a protected timestamp record at the earliest timestamp
+	// that the fingerprint query is going to read from. When fingerprinting
+	// revisions, this will be the specified start time.
+	tsToProtect := params.p.EvalContext().Txn.ReadTimestamp()
+	if n.options != nil && !n.options.startTimestamp.IsEmpty() {
+		tsToProtect = n.options.startTimestamp
+	}
 	cleanup, err := protectTenantSpanWithSession(
 		params.ctx,
-		params.p.EvalContext(),
 		params.p.ExecCfg(),
 		tid,
 		params.p.ExtendedEvalContext().SessionID,
+		tsToProtect,
 	)
 	if err != nil {
 		return false, err
 	}
 	defer cleanup()
 
+	var startTime hlc.Timestamp
+	var allRevisions bool
+	if n.options != nil && !n.options.startTimestamp.IsEmpty() {
+		startTime = n.options.startTimestamp
+		allRevisions = true
+		n.run.values[1] = eval.TimestampToDecimalDatum(startTime)
+	}
+
 	fingerprint, err := params.p.FingerprintSpan(params.ctx,
 		keys.MakeTenantSpan(tid),
-		hlc.Timestamp{},
-		false,
-		true)
+		startTime,
+		allRevisions,
+		!allRevisions /* stripped */)
```

Comment:
I think we still want this as false? (note that FingerprintSpans _always_ strips the tenant prefix and value checksum, this is about whether we strip timestamps and index prefixes and I think we never really want to strip the latter.

Reply:
> I think we still want this as false?
I'm a little confused, `stripped` was prior to this patch set to `true`. Did we mean to set it to false instead? I did for a second think that this looked fishy but got distracted by other stuff.

Reply:
Ah, yeah....I think it was perhaps wrong before..

Reply:
gotcha, changed to false.